cfn/templates/service.template.yml

```diff
@@ -0,0 +1,1754 @@
+        {{ allow_destroy() }}
+        Type: AWS::EC2::SecurityGroupIngress
+        Properties:
+            Description: Allow ALB to reach ECS for {{ global.label }}/{{ router.alias }}
+            SourceSecurityGroupId: !Ref LoadBalancerSecurityGroup{{ router.alias }}
+            IpProtocol: tcp
+            FromPort: 32768
+            ToPort: 65535
+            GroupId: !Sub "${ClusterHostSecurityGroup}"
+    {%- endif %}
+
+    LoadBalancerListener{{ router.alias }}:
+        {{ allow_destroy() }}
+        Type: AWS::ElasticLoadBalancingV2::Listener
+        DependsOn:
+          - LoadBalancer{{ router.alias }}
+          - DefaultTargetGroup{{ router.alias }}
+        Properties:
+            LoadBalancerArn: !Ref LoadBalancer{{ router.alias }}
+            Port: {{ router.public_port }}
+            Protocol: {{ router.public_protocol | upper }}
+          {%- if router.certificate is defined %}
+            Certificates:
+              - CertificateArn: {{ router.certificate }}
+            SslPolicy: ELBSecurityPolicy-TLS-1-2-Ext-2018-06
+          {%- endif %}
+            DefaultActions:
+              - Type: forward
+                TargetGroupArn: !Ref DefaultTargetGroup{{ router.alias }}
+
+    {%- if router.http_to_https_redirect %}
+
+    LoadBalancerListener{{ router.alias }}Redirect:
+        {{ allow_destroy() }}
+        Type: AWS::ElasticLoadBalancingV2::Listener
+        DependsOn:
+          - LoadBalancer{{ router.alias }}
+          - DefaultTargetGroup{{ router.alias }}
+        Properties:
+            LoadBalancerArn: !Ref LoadBalancer{{ router.alias }}
+            Port: 80
+            Protocol: HTTP
+            DefaultActions:
+              - Type: redirect
+                RedirectConfig:
+                    Host: "#{host}"
+                    Path: "/#{path}"
+                    Port: "443"
+                    Protocol: "HTTPS"
+                    Query: "#{query}"
+                    StatusCode: HTTP_301
+    {%- endif %}
+
+    DefaultTargetGroup{{ router.alias }}:
+        {{ allow_destroy() }}
+        Type: AWS::ElasticLoadBalancingV2::TargetGroup
+        Properties:
+          {%- if router.target_group_name is defined %}
+            Name: {{ router.target_group_name }}
+          {%- endif %}
+
+            VpcId: {{ global.vpc_import }}
+            Port: {{ router.container_port }}
+            Protocol: {{ router.container_protocol | upper }}
+
+          {%- if router.perform_health_check %}
+            HealthCheckPath: {{ router.health_check.path }}
+            Matcher: {{ router.health_check.matcher }}
+            HealthCheckIntervalSeconds: {{ router.health_check.interval_seconds }}
+            HealthyThresholdCount: {{ router.health_check.healthy_threshold_count }}
+            UnhealthyThresholdCount: {{ router.health_check.unhealthy_threshold_count }}
+            HealthCheckTimeoutSeconds: {{ router.health_check.timeout_seconds }}
```

Comment:
In order for the deployed `AWS::ElasticLoadBalancingV2::TargetGroup` to be altered, I think we would need to also change this section of code, as this is the section of code which actually determines the config of the resource.